core/windows/src/internal/TzdbInRegistry.kt

```diff
@@ -60,7 +60,16 @@ internal class TzdbInRegistry: TimeZoneDatabase {
                         }
                     }
                 }
-                if (offsets.isEmpty()) { offsets.add(recurring.offsetAtYearStart()) }
+                offsets.lastOrNull()?.let { lastOffset ->
+                    /* If there are already some offsets, we can not add a new offset without defining a transition to
+                    it. The moment when we start using the recurring rules is the first year that does not have any
+                    historic data provided. */
+                    val firstYearWithRecurringRules = historic.last().first + 1
+                    val newYearInLastOffset = LocalDate(firstYearWithRecurringRules, Month.JANUARY, 1).atTime(0, 0)
+                        .toInstant(lastOffset)
+                    transitionEpochSeconds.add(newYearInLastOffset.epochSeconds)
+                }
+                offsets.add(recurring.offsetAtYearStart())
```

Comment:
Can it create a transition from an offset to the same offset if added unconditionally?

Reply:
Yes, but I can't think of a way to see the extra transitions using the API exposed to our users.

Reply:
Ok, let's think about it if/when we're going to expose transition rules later